### PR TITLE
feat: Add app link for password reset

### LINKED ISSUE #1796: Add app link for password reset
**Describe the feature you'd like**
Add app link for password reset.

**Would you like to work on the issue?**
- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by ShridharGoel**
Fixes #1796

### CODE REVIEW

File: app/src/main/java/com/eventyay/organizer/core/auth/AuthActivity.java [hunk 1]
```diff
@@ -41,6 +44,26 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
                 .replace(R.id.fragment_container, new StartFragment())
                 .commit();
         }
+
+        handleIntent(getIntent());
+    }
+
+    protected void onNewIntent(Intent intent) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/AuthActivity.java]: @Override

File: app/src/main/java/com/eventyay/organizer/core/auth/AuthActivity.java [hunk 2]
```diff
@@ -41,6 +44,26 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
...
+        super.onNewIntent(intent);
+        handleIntent(intent);
+    }
+
+    private void handleIntent(Intent intent) {
+        String appLinkAction = intent.getAction();
+        Uri appLinkData = intent.getData();
+        if (Intent.ACTION_VIEW.equals(appLinkAction) && appLinkData != null){
+            String token = appLinkData.toString().substring(
+                appLinkData.toString().indexOf('=') + 1);
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/auth/AuthActivity.java]: Add a separate class to parse the link and add tests

File: app/src/main/java/com/eventyay/organizer/core/auth/AuthActivity.java (hunk 1 above)
Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/auth/AuthActivity.java]: Updated

File: app/src/main/java/com/eventyay/organizer/core/auth/AuthActivity.java (hunk 2 above)
Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/core/auth/AuthActivity.java]: Done.

File: app/src/main/java/com/eventyay/organizer/utils/LinkUtils.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.eventyay.organizer.utils;
+
+public class LinkUtils {
+
+    private LinkUtils() {
+        // Never Called
+    }
+
+    public static String getToken(String url) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/utils/LinkUtils.java]: What will you do when you have to handle link to verify password?

Create an enum Destination with identifier and return it

```kotlin
enum Destination {
  PASSWORD_VERIFY,
  PASSWORD_RESET
}

class DestinationWithIdentifier(destination, identifier: String)

DestinationWithIdentifier getDestination(String url) {
  if (url is password reset)
    return DestinationWithIdentifier(PASSWORD_RESET, token)
  else if (url is password verify)
    return DestinationWithIdentifier(PASSWORD_VERIFY, token)
  else
    return null
}
```

Comment (ShridharGoel) [reply app/src/main/java/com/eventyay/organizer/utils/LinkUtils.java]: Updated.

**Pull request merged**